### PR TITLE
12431 Better user experience for final dot critique and cleaning

### LINKED ISSUE #12431: Better user experience for final dot critique and cleaning
By default we would like to remove unnecessary separators from method body to istay clean.

See 
    https://twitter.com/TorstenAstares/status/1609666189382254593
    https://twitter.com/TorstenAstares/status/1609952076804628482
and #12387

This includes unnecessary and unwanted spaces, tabs but also final dots at the end of the method.

Despite the cleaning and the default in the Pharo standard image some users might nonetheless
want to continue with them for their own packages.  They also might not want to be criticized for 
their the final dots.

So we should increase the user experience by providing a setting where they can disable this
default in the settings browser like:

image

### PR DESCRIPTION
**Pull request by astares**
Fix #12431

### CODE REVIEW

File: src/Renraku-Tests/ReMethodSourceCleanerTest.class.st [hunk 1]
```diff
@@ -7,8 +7,117 @@ Class {
...
+
+{ #category : #'private - test data' }
+ReMethodSourceCleanerTest >> alreadyCleanedMethodString [
+	^'dirtyMethodWithUnnecessarySpacesAndFinalDot
+	"Do not format this method - it by intention should include unnecessary spaces and a final dot"
+
+	|temp|
+
+	temp := 3 + 4'
+]
```
Comment (jordanmontt) [inline src/Renraku-Tests/ReMethodSourceCleanerTest.class.st:32]: It looks like you have this method duplicated

File: src/Renraku-Tests/ReMethodSourceCleanerTest.class.st [hunk 2]
```diff
@@ -7,8 +7,117 @@ Class {
...
+	|temp|
+
+	temp := 3 + 4
+]
+
+{ #category : #'private - test data' }
+ReMethodSourceCleanerTest >> properMethodWithFinalDot [
+	"This is a method without any unnecessary separators in the method body but a final dot"
+
+	^ 3 + 4.
```
Comment (jordanmontt) [inline src/Renraku-Tests/ReMethodSourceCleanerTest.class.st:47]: Maybe add a warning in the comment to prevent people to format this method

**Review by jordanmontt — APPROVED** [review]

File: src/Renraku-Tests/ReMethodSourceCleanerTest.class.st (hunk 1 above)
Comment (astares) [reply src/Renraku-Tests/ReMethodSourceCleanerTest.class.st:32]: No - as you see and as the selector should reveal: one returns the cleand method with a dot and one without. It's also not an own method body - it returns a string we compare agains the #dirtyMethodWithUnnecessarySpacesAndFinalDot in the tests

File: src/Renraku-Tests/ReMethodSourceCleanerTest.class.st (hunk 2 above)
Comment (astares) [reply src/Renraku-Tests/ReMethodSourceCleanerTest.class.st:47]: Made it more clear in https://github.com/astares/pharo/commit/4904ac8610c2f07e369be13a25efb174b07400ba

File: src/Renraku-Tests/ReMethodSourceCleanerTest.class.st (hunk 1 above)
Comment (astares) [reply src/Renraku-Tests/ReMethodSourceCleanerTest.class.st:32]: But I added comment in https://github.com/astares/pharo/commit/4904ac8610c2f07e369be13a25efb174b07400ba
too

Comment (jordanmontt) [reply src/Renraku-Tests/ReMethodSourceCleanerTest.class.st:32]: Didn't see the dot at the end :)

Comment (astares) [reply src/Renraku-Tests/ReMethodSourceCleanerTest.class.st:32]: That's why ideally there should be none :)

**Comment by astares** [comment]
CI failing tests unrelated

**Review by guillep — APPROVED** [review]

**Comment by guillep** [comment]
Thanks!

**Pull request merged**